src/crystal/spin_lock.cr

```diff
@@ -0,0 +1,40 @@
+module Crystal
+  # :nodoc:
+  #
+  # Spin lock that prevents threads from executing a critical section at the
+  # same time. Since it blocks a thread from progressing, this should only be
+  # used for fast thread synchronisation primitives.
+  struct SpinLock
+    def initialize
+      @flag = Atomic::Flag.new
+    end
+
+    # Tries to acquire the lock or blocks the current operating system thread
+    # until the lock is acquired.
+    def lock : Nil
+      # fast path: always succeeds with a single thread:
+      until try_lock
+        # fixed busy loop, avoids a thread context switch which improves
+        # performance with many parallel threads:
+        99.times do |i|
```

Comment:
Ditto. If there are no other option, it can be made a constant shared with the other previous `99.times`.

Reply:
No. This SpinLock is the generic struct used by Mutex and ConditionVariable (checks an atomic flag). Despite looking the same, it's different from the resumable spin lock (checks an integer). Using a constant shared between both objects would introduce an artificial dependency: one doesn't need the other.

Reply:
It seems to work well with this number, I don't see the point of moving it to a constant. Then to understand the code you must jump between places.

Reply:
Why would you have to jump between places to understand the code? Just to get the number? Usually the constant name should explain it's purpose.